nipype/algorithms/compcor.py

```diff
@@ -0,0 +1,81 @@
+from ..interfaces.base import (BaseInterfaceInputSpec, TraitedSpec,
+                               BaseInterface, traits, File)
+import nibabel as nb
+import numpy as np
+from scipy import linalg, stats
+import os
+
+from nipype.pipeline.engine import Workflow
+
+class CompCoreInputSpec(BaseInterfaceInputSpec):
+    realigned_file = File(exists=True, mandatory=True, desc='already realigned brain image (4D)')
+    mask_file = File(exists=True, mandatory=True, desc='mask file that determines ROI (3D)')
+    num_components = traits.Int(6, usedefault=True) # 6 for BOLD, 4 for ASL
+    # additional_regressors??
+
+class CompCoreOutputSpec(TraitedSpec):
+    components_file = File(desc='text file containing the noise components', exists=True)
+
+class CompCore(BaseInterface):
```

Comment:
Yup! It's just a question of adding `references_` field such as here:
https://github.com/nipy/nipype/blob/0f2cb9fce47dc10cfb29be7002193481c876f8e0/nipype/interfaces/spm/base.py#L239

On Tue, Sep 6, 2016 at 2:15 PM, Shoshana Berleant <notifications@github.com>
wrote:

> In nipype/algorithms/compcor.py
> <https://github.com/nipy/nipype/pull/1599#discussion_r77720814>:
>
> > +import numpy as np
> > +from scipy import linalg, stats
> > +import os
> > +
> > +from nipype.pipeline.engine import Workflow
> > +
> > +class CompCoreInputSpec(BaseInterfaceInputSpec):
> > +    realigned_file = File(exists=True, mandatory=True, desc='already realigned brain image (4D)')
> > +    mask_file = File(exists=True, mandatory=True, desc='mask file that determines ROI (3D)')
> > +    num_components = traits.Int(6, usedefault=True) # 6 for BOLD, 4 for ASL
> > +    # additional_regressors??
> > +
> > +class CompCoreOutputSpec(TraitedSpec):
> > +    components_file = File(desc='text file containing the noise components', exists=True)
> > +
> > +class CompCore(BaseInterface):
>
> Yes it's a pun :) I'll change it.
>
> re doi's, is this what you're referring to? #1464
> <https://github.com/nipy/nipype/issues/1464>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/nipy/nipype/pull/1599/files/ef50858d3b3016018f0078af6b7da29412803c00#r77720814>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOkp7UyOpbrANACeVwGabDKi06T-Wveks5qndgEgaJpZM4JyQmp>
> .
>